### DISCUSSION TITLE
STM32F4x App not starting

### DISCUSSION THREAD

**Discussion by vibr77** [adafruit/tinyuf2#416] [Q&A]
Hello, 
Thanks for supporting tinyUF2. 

I have a STM32F411 with the tinyUF2 boot loader successfully installer.
the device enters in USB mode with double tap on RST button and using UART I can see logs. 

I have an application firmware that is correctly working without tinyUF2. but it is not starting with UF2 format.

I have done the following  :
- modified system_stm32f4xx.c:

add #define USER_VECT_TAB_ADDRESS 
changed #define VECT_TAB_OFFSET         0x00010000U

changed the linker file STM32F411CEUx_FLASH.ld:

`MEMORY
{
  RAM (xrw)     : ORIGIN = 0x20000000, LENGTH = 128K - 4 /* reserve 4 bytes for double tap   */
  FLASH (rx)    : ORIGIN = 0x08010000, LENGTH = 448K    /* must match BOARD_FLASH_APP_START */
  CONFIG (rx)   : ORIGIN = 0x08008000 - 1024, LENGTH = 1024
}`

I use the following command to build a UF2 file based on the HEX file

python3 uf2conv.py -c -b 0x08010000 -f 0x57755a57 firmware.hex -o ./build/firmware.UF2

Then I upload the firmware.UF2 over USB file drap & drop. 

I can see in the UART the upload and as well the handover to the App but after that nothing. it is the same with the blinking demo app.

How could I solve this ? What do I need to check ? 

I have to admit that I am a bit lost

Attached the boot loader & app file UF2
[AppleIISDiskII_stm32f411.UF2.zip](https://github.com/user-attachments/files/17449435/AppleIISDiskII_stm32f411.UF2.zip)
[tinyuf2-stm32f411ce_blackpill.elf.zip](https://github.com/user-attachments/files/17449437/tinyuf2-stm32f411ce_blackpill.elf.zip)

**Comment by hathach** [comment]
it is probably due to the VTOR is set to 0x0 by system_stm32xxx.c try to add this line to your main() https://github.com/adafruit/tinyuf2/pull/419/files#diff-19aa9dc949d3bb8855bb23418f05fc68134dcbc6a74aa6a63a4f311d8a3adc45R67 . I will revise and add blinky app for testing for this port later.
